enhancements/installer/azure-support-known-cloud-environments.md

```diff
@@ -0,0 +1,231 @@
+
+#### Installer
+
+##### Install Config
+
+The install config should allow the user to set the cloud name for Azure.
+
+```go
+// types/pkg/azure/platform.go
+type Platform struct {
+    ...
+	// cloudName is the name of the Azure cloud environment which can be used to configure the Azure SDK
+	// with the appropiate Azure API endpoints.
+	// If empty, the value is equal to `AzurePublicCloud`.
+	// +optional
+	CloudName AzureCloudEnvironment `json:"cloudName,omitempty"`
+}
+
+// +kubebuilder:validation:Enum="";AzurePublicCloud;AzureUSGovernmentCloud;AzureChinaCloud;AzureGermanCloud
+type AzureCloudEnvironment string
+
+const (
+	PublicAzureCloud AzureCloudEnvironment = "AzurePublicCloud"
+	USGovernmentAzureCloud AzureCloudEnvironment = "AzureUSGovernmentCloud"
+	AzureChinaCloud AzureCloudEnvironment = "AzureChinaCloud"
+	AzureGermanCloud AzureCloudEnvironment = "AzureGermanCloud"
+)
+```
+
+##### Configuring the Azure SDK
+
+The installer uses the Authorizers generated from [`newSessionFromFile`](https://github.com/openshift/installer/blob/a11cc4d29231135cd889487275ee32a20546b7e2/pkg/asset/installconfig/azure/session.go#L50) to create clients for Azure APIs. Updating the function to accept Azure cloud environment name should allow correct initiation of the clients.
+
+##### Configuring the internal terraform-provider-azurerm with environment
+
+The AzureRM provider accepts [`environment`](https://www.terraform.io/docs/providers/azurerm/index.html#environment) field to configure the Azure SDK. The valid values are `public`, `usgovernment`, `german`, and `china`, the installer should convert the install-config.yaml `AzureCloudEnvironment` string to a value acceptable by the provider.
+
+##### Configuring infrastructure object
+
+The installer should transform the install-config.yaml `AzureCloudEnvironment` string to set the `.status.platformStatus.azure.cloudName` in the Infrastructure object as detailed [above](#Infrastructure-global-configuration)
+
+##### Destroy
+
+The metadata.json needs to store the cloud name from the install-config.yaml so that the users can delete the clusters without any previous state.
+
+#### Cluster Operators
+
+Almost of the cluster operators will have to read the cloud name from the Infrastructure's object's status for Azure and make sure the operators are watching for changes.
+
+The cluster operators using the Azure Go SDK can utilize [`EnvironmentFromName`](https://github.com/Azure/go-autorest/blob/9132adfc9db3007653dccb091b6a6c8e09b74ef5/autorest/azure/environments.go#L211-L230) to load the Azure cloud environment by passing the `.status.platformStatus.azure.cloudName` as string to the function.
+
+The cluster operators must make sure to use `AzurePublicCloud` as value when `.status.platformStatus.azure.cloudName` is empty.
+
+##### Machine API
```

Comment:
cc @openshift/openshift-team-cloud

Reply:
ping @enxebre @michaelgugino for cloud

Reply:
sgtm for the controller to fetch this from the infra API.